libp2p/network/url_conversion.go

```diff
@@ -35,25 +41,38 @@ func Libp2pMultiaddrFromServiceURL(serviceURL string) (multiaddr.Multiaddr, erro
 		peerIPVersionStr = ipVersion4
 	)
 
-	peerUrl, err := url.Parse(anyScheme + serviceURL)
+	// Conditionally add the anyScheme prefix if no scheme is present.
+	if strings.Contains(serviceURL, "://") {
+		return nil, fmt.Errorf("usage of scheme is invalid service URL: %s", serviceURL)
+	} else {
+		serviceURL = anyScheme + serviceURL
+	}
+
+	peerUrl, err := url.Parse(serviceURL)
 	if err != nil {
-		return nil, fmt.Errorf(
-			"parsing peer service URL: %s: %w",
-			serviceURL,
-			err,
-		)
+		return nil, fmt.Errorf("parsing peer service URL: %s: %w", serviceURL, err)
+	}
+
+	// Check if service URL contains a port number
+	if _, port, err := net.SplitHostPort(peerUrl.Host); err != nil || port == "" {
+		return nil, fmt.Errorf("missing port number and delimiter in service URL: %s", serviceURL)
```

Comment:
Let's move the message (up till `:`) into a new error message variable please (see below)